utils/src/test/scala/cromwell/pipeline/utils/TestContainersUtils.scala

```diff
@@ -2,30 +2,43 @@ package cromwell.pipeline.utils
 
 import com.dimafeng.testcontainers.PostgreSQLContainer
 import com.typesafe.config.{ Config, ConfigFactory }
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.containers.wait.strategy.Wait
 
 object TestContainersUtils {
 
-  private val dbConfig = ConfigFactory.load().getConfig("database.postgres_dc.db.properties")
-  private val user = dbConfig.getString("user")
-  private val databaseName = dbConfig.getString("databaseName")
-  private val password = dbConfig.getString("password")
-  private val port = dbConfig.getInt("portNumber")
+  private lazy val postgreConfig = ConfigFactory.load().getConfig("database.postgres_dc.db.properties")
 
   def getPostgreSQLContainer(postgresImageName: String = "postgres:12"): PostgreSQLContainer =
     PostgreSQLContainer(
       dockerImageNameOverride = postgresImageName,
-      databaseName = databaseName,
-      username = user,
-      password = password
+      databaseName = postgreConfig.getString("databaseName"),
+      username = postgreConfig.getString("user"),
+      password = postgreConfig.getString("password")
     )
 
+  def getConfigForPgContainer(container: PostgreSQLContainer): Config =
+    config(
+      "database.postgres_dc.db.properties.portNumber" ->
+        container.mappedPort(postgreConfig.getInt("portNumber"))
+    )
+
+  private lazy val mongoConfig = ConfigFactory.load().getConfig("database.mongo")
+  private def mongoPort: Int = mongoConfig.getInt("port")
+
+  class MongoContainer(dockerImageName: String) extends GenericContainer[MongoContainer](dockerImageName)
+
+  def getMongoContainer(mongoImageName: String = "mongo"): MongoContainer =
+    new MongoContainer(mongoImageName)
+      .withExposedPorts(mongoPort)
+      .withCommand("--replSet", "docker-rs")
+      .waitingFor(Wait.forLogMessage(".*Waiting for connections.*", 1))
+
+  def getConfigForMongoContainer(container: GenericContainer[_]): Config =
+    config("database.mongo.port" -> container.getMappedPort(mongoPort))
```

Comment:
These two methods are unused. Is this intentional?

Reply:
Yes, my idea was to add mongo container provider/infrastructure for future